sky/data/storage.py

```diff
@@ -986,6 +991,62 @@ def sync_all_stores(self):
         for _, store in self.stores.items():
             self._sync_store(store)
 
+    def _compress_sync_store(self, store: AbstractStore):
+        """Same as sync_store, but compresses before uploading"""
+        if self.source is None:
+            self._sync_store(store)
+            return
+        zip_filepath = os.path.join(os.getcwd(), f'{store.name}-compressed')
+        zip_filename = os.path.join(zip_filepath, 'skypilot-filemounts.zip')
+        filepaths = self.source
+        if isinstance(self.source, str):
+            filepaths = [self.source]
+        try:
+            if not os.path.exists(zip_filepath):
+                os.mkdir(zip_filepath)
+            with zipfile.ZipFile(os.path.join(zip_filepath, zip_filename), 'w',
+                                 zipfile.ZIP_DEFLATED) as zipf:
+                for filepath in filepaths:
+                    if os.path.isdir(os.path.expanduser(filepath)):
+                        for root, _, files in os.walk(filepath):
+                            for file in files:
+                                if file == 'skypilot-filemounts.zip':
+                                    continue
+                                zipf.write(
+                                    os.path.join(root, file),
+                                    os.path.join(root.replace(filepath, '', 1),
+                                                 file) + '/')
+                    else:
+                        logger.info(f'Filepath Zipping {filepath}')
+                        zipf.write(filepath, filepath.replace(os.path.sep, '/'))
+        except:
+            if os.path.exists(zip_filename):
+                os.remove(zip_filename)
+                os.rmdir(zip_filepath)
+            raise
```

Comment:
Since we want to include exclude files as specified in the gitignore, it may be equally or more concise to do it this way (make_archive lacks a way to ignore certain files, although we could do copy tree and then do make_archive but I think this would end up being more complicated)

Reply:
Can we use `--exclude` for `tar` or `-x` for `zip`?

Reply:
That makes sense, I will go with that approach then